hazelcast/src/hazelcast/util/impl/Executor.cpp

```diff
@@ -189,27 +232,32 @@ namespace hazelcast {
                     thread(boost::shared_ptr<util::Runnable>(new util::RunnableDelegator(*this)), executorService.logger) {
             }
 
-            SimpleExecutorService::RepeatingRunner::RepeatingRunner(
+            Thread &SimpleExecutorService::Worker::getThread() {
+                return thread;
+            }
+
+            SimpleExecutorService::RepeatingRunner::RepeatingRunner(SimpleExecutorService &executorService,
                     const boost::shared_ptr<util::Runnable> &command, int64_t initialDelayInMillis,
-                    int64_t periodInMillis, util::ILogger &logger) : DelayedRunner(command, initialDelayInMillis, periodInMillis, logger) {
+                    int64_t periodInMillis, util::ILogger &logger) : DelayedRunner(executorService, command,
+                            initialDelayInMillis, periodInMillis, logger) {
             }
 
             const std::string SimpleExecutorService::RepeatingRunner::getName() const {
                 return command->getName();
```

Comment:
It seems that we are passing the executorService for `threadNamePrefix` but it is forgotten to be used here. 
And if the only purpose is `threadNamePrefix`, we should only pass that on the constructor.

Reply:
Deleted RepeatingRunner class.